client/components/domains/connect-domain-step/transfer-domain-step-unlock.jsx

```diff
@@ -1,24 +1,103 @@
 import { Button } from '@automattic/components';
 import { useI18n } from '@wordpress/react-i18n';
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import CardHeading from 'calypso/components/card-heading';
 import ConnectDomainStepWrapper from 'calypso/components/domains/connect-domain-step/connect-domain-step-wrapper';
+import {
+	stepsHeadingTransfer,
+	domainLockStatusType,
+} from 'calypso/components/domains/connect-domain-step/constants';
+import MaterialIcon from 'calypso/components/material-icon';
+import Notice from 'calypso/components/notice';
+import wpcom from 'calypso/lib/wp';
 
-const TransferDomainStepUnlock = ( { className, onNextStep, ...props } ) => {
+const TransferDomainStepUnlock = ( {
+	className,
+	onNextStep,
+	domain,
+	domainLockStatus: initialDomainLockStatus,
+	...props
+} ) => {
 	const { __ } = useI18n();
 
+	const [ checkInProgress, setCheckInProgress ] = useState( false );
+	const [ domainStatusError, setDomainStatusError ] = useState( null );
+
+	const getDomainLockStatus = useCallback( async () => {
+		setCheckInProgress( true );
+		const { unlocked } = await wpcom.undocumented().getInboundTransferStatus( domain );
+		setCheckInProgress( false );
+		return unlocked;
+	}, [ domain, setCheckInProgress ] );
+
+	const checkDomainLockStatus = async () => {
+		try {
+			const isDomainUnlocked = await getDomainLockStatus();
+			if ( isDomainUnlocked ) onNextStep();
+			else {
+				setDomainStatusError( 'Your domain is locked' );
+			}
+		} catch {
+			setDomainStatusError( 'Can’t get the domain’s lock status' );
+		}
+	};
+
+	const lockedDomainContent =
+		initialDomainLockStatus === domainLockStatusType.UNLOCKED ? null : (
+			<CardHeading tagName="h2" className={ className + '__sub-heading' }>
+				<MaterialIcon className={ className + '__sub-heading-icon' } size={ 24 } icon="lock" />
+				{ initialDomainLockStatus === domainLockStatusType.LOCKED
+					? __( 'Your domain is locked' )
+					: __( "Can't get the domain's lock status" ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'Can't get the domain's lock status.' )` *ES Score: 12*